pkg/daemon/writer.go

```diff
@@ -242,8 +261,17 @@ func (nw *clusterNodeWriter) Eventf(eventtype, reason, messageFmt string, args .
 	nw.recorder.Eventf(getNodeRef(nw.node), eventtype, reason, messageFmt, args...)
 }
 
-func implSetNodeAnnotations(client corev1client.NodeInterface, lister corev1lister.NodeLister, nodeName string, m map[string]string) response {
+func implSetNodeAnnotations(client corev1client.NodeInterface, lister corev1lister.NodeLister, nodeName string, m map[string]string, toDel []string) response {
 	node, err := internal.UpdateNodeRetry(client, lister, nodeName, func(node *corev1.Node) {
+		if toDel != nil {
```

Comment:
nit: I think this line is technically not required as the for loop would do nothing if `toDel` is empty, but no harm in having it here

Reply:
I've been burned by nil slices and maps in the past, so that `if toDel != nil` is a reflex-action at this point :smile:. I'll leave it for now though.